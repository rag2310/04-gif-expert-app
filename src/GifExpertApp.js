import React, {useState} from 'react';
import {AddCartegory} from './components/AddCartegory';
import {GifGrid} from './components/GifGrid';

const GifExpertApp = ({defaultCategories = []}) => {

    // const [categories, setCategories] = useState(['One Punch']);
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCartegory setCategories={setCategories}/>
            <hr/>

            <ol>
                {
                    categories.map(category => {
                        return (

                            <GifGrid
                                key={category}
                                category={category}
                            />
                        );
                    })
                }
            </ol>
        </>
    );
}


export default GifExpertApp;