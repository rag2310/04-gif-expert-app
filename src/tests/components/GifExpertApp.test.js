import {shallow} from "enzyme";
import '@testing-library/jest-dom';
import GifExpertApp from "../../GifExpertApp";

describe('Pruebas de <GifExpertApp />', function () {

    test('debe mostrar <GifExpertApp />', () => {

        const wrapper = shallow(<GifExpertApp/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostrar una lista de categorias', () => {

        const categories = ['One puch', 'Dragon ball']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
});