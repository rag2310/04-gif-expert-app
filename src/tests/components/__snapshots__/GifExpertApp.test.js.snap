// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifExpertApp /> debe mostrar <GifExpertApp /> 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCartegory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas de <GifExpertApp /> debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCartegory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One puch"
      key="One puch"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;
